tests/sources/test_redis.py

```diff
@@ -0,0 +1,370 @@
+
+DOCUMENT = [
+    {
+        "_id": "aa00c4c0f44c5cb7cad68df40e8f8877",
+        "key": "0",
+        "value": "this is value",
+        "size_in_bytes": 10,
+        "database": 0,
+        "key_type": "string",
+        "_timestamp": "2023-01-24T04:07:19+00:00",
+    }
+]
+
+
+class RedisClientMock:
+    async def execute_command(self, SELECT="JSON.GET", key="json_key"):
+        return json.dumps({"1": "1", "2": "2"})
+
+    async def zrange(self, key, start, skip, withscores=True):
+        return {1, 2, 3}
+
+    async def smembers(self, key):
+        return {1, 2, 3}
+
+    async def get(self, key):
+        return "this is value"
+
+    async def hgetall(self, key):
+        return "hash"
+
+    async def xread(self, key):
+        return "stream"
+
+    async def lrange(self, key, start, skip):
+        return [1, 2, 3]
+
+    async def config_get(self, databases):
+        return {"databases": "1"}
+
+    async def ping(self):
+        return False
+
+    async def aclose(self):
+        return True
+
+    async def type(self, key):  # NOQA
+        return "string"
+
+    async def memory_usage(self, key):
+        return 10
+
+    async def scan_iter(self, match, count, _type):
+        yield "0"
+
+    async def validate_database(self, db=0):
+        await self.execute_command()
+        await self.ping()
+
+
+@asynccontextmanager
+async def create_redis_source():
+    async with create_source(
+        RedisDataSource,
+        host="localhost",
+        port=6379,
+        database="0",
+        username="username",
+        password="password",
+    ) as source:
+        yield source
+
+
+@pytest.mark.asyncio
+async def test_ping_positive():
+    async with create_redis_source() as source:
+        source.client.ping = AsyncMock()
+        await source.ping()
+
+
+@pytest.mark.asyncio
+async def test_ping_negative():
+    async with create_redis_source() as source:
+        mocked_client = Mock()
+        with mock.patch("redis.from_url", return_value=mocked_client):
+            mocked_client.ping = AsyncMock(
+                side_effect=redis.exceptions.AuthenticationError
+            )
+            with pytest.raises(Exception):
+                await source.ping()
+
+
+@pytest.mark.asyncio
+async def test_validate_config_when_database_type():
+    async with create_redis_source() as source:
+        source.client.database = ["1", "db123", "123"]
+        mocked_client = Mock()
+        mocked_client.validate_database = AsyncMock(return_value=True)
+        with mock.patch("redis.from_url", return_value=mocked_client):
+            with pytest.raises(ConfigurableFieldValueError):
+                await source.validate_config()
+
+
+@pytest.mark.asyncio
+async def test_validate_config_when_database_is_invalid():
+    async with create_redis_source() as source:
+        source.client.database = ["123"]
+        mocked_client = Mock()
+        mocked_client.validate_database = AsyncMock(return_value=True)
```

Comment:
Same here, I'm a bit confused, why we return `True`, which indicates a valid database, but the test states that the database is invalid? Could you elaborate here?